franmolmedo/react-context-and-portals/readme.md

```diff
@@ -0,0 +1,298 @@
+
+*   Componente App:
+
+```javascript
+import React, { Component } from 'react';
+import Counter from './Counter';
+
+class App extends Component {
+    state = {
+        current: 0
+    };
+
+    render() {
+        return (
+            <Counter
+                currentValue={this.state.current}
+                actions={{
+                    increment: () =>
+                        this.setState({ current: this.state.current + 1 }),
+                    decrement: () =>
+                        this.setState({ current: this.state.current - 1 })
+                }}
+            />
+        );
+    }
+}
+
+export default App;
+```
+
+*   Componente Counter:
+
+```javascript
+import React, { Fragment } from 'react';
+
+import CounterValue from './CounterValue';
+
+export default ({ currentValue, actions }) => (
+    <Fragment>
+        <CounterValue currentValue={currentValue} />
+        <button onClick={() => actions.increment()}>INCREMENT</button>
+        <button onClick={() => actions.decrement()}>DECREMENT</button>
+    </Fragment>
+);
+```
+
+*   Componente CounterValue:
+
+```javascript
+import React from 'react';
+
+export default ({ currentValue }) => (
+    <h1>The current value is: {currentValue}</h1>
+);
+```
+
+En este caso vemos que el valor actual del contador tiene que llegar desde el componente **_App_** hasta el componente **_CounterValue_** pasando por **_Counter_**. Todavía en este caso sería manejable este paso de elementos por **_props_**, pero podemos extrapolarlo a un caso en el que hayan 6, 7 o 10 niveles intermedios y ya no sería tan manejable. Igualmente, hemos remodelado el state para definir las funciones dentro del mismo y pasarlas a los componentes hijos de forma más sencilla.
+
+#### Context API al rescate
+
+En un caso como el comentado, quizás se nos ocurra utilizar una librería externa para el manejo del estado de la aplicación (como [redux](https://github.com/reactjs/redux)) conectaría mis componentes a la store utilizando el **HOC** **_connect_** de [react-redux](https://github.com/reactjs/react-redux) para mapear las propiedades de nuestro estado como prop del componente deseado.
+Con **_context_** el procedimiento sería parecido. En primer lugar, generaremos nuestro contexto para mapear nuestra store:
+
+![creating-context](images/creating-context.gif).
+
+El código final de nuestro conexto quedará de la siguiente forma:
+
+```javascript
+import React, { Component } from 'react';
+
+export const AppContext = new React.createContext();
+
+export class AppProvider extends Component {
+    state = {
+        current: 0
+    };
+
+    increment = () => this.setState({ current: this.state.current + 1 });
+    decrement = () => this.setState({ current: this.state.current - 1 });
+
+    render() {
+        const value = {
+            state: this.state,
+            actions: {
+                increment: this.increment.bind(this),
```

Comment:
why not use here
actions: {
    increment: () => this.increment(),
    decrement: () => this.decrement()
}

like previous steps?

Reply:
Showing the two ways to do the same.